src/main/haskell/language-kore/app/format/Main.hs

```diff
@@ -0,0 +1,56 @@
+module Main where
+
+import         Data.Semigroup ((<>))
+import         Data.Text.Prettyprint.Doc ( LayoutOptions(..), PageWidth(..)
+                                         , defaultLayoutOptions, layoutPretty
+                                         , pretty )
+import         Data.Text.Prettyprint.Doc.Render.Text (renderIO)
+import         Options.Applicative
+import         System.IO (stdout)
+
+import         Data.Kore.Parser.Parser (fromKore)
+
+import         GlobalMain
+
+
+data KoreFormatOptions =
+    KoreFormatOptions
+    { fileName :: FilePath  -- ^ file to unparse
+    , width :: Int  -- ^ line width
+    }
+
+commandLine :: Parser KoreFormatOptions
+commandLine =
+    KoreFormatOptions
+    <$> argument str
+        (  metavar "FILE"
+        <> help "Kore source file to parse" )
+    <*> option auto
+        (  metavar "WIDTH"
+        <> long "width"
+        <> value 80
+        <> help "Line width [default: 80]" )
+
+infoMod :: InfoMod options
+infoMod =
+    fullDesc
+    <> progDesc "Parse a Kore definition and render it in standard format"
+    <> header "kore-format - parse and render Kore definitions"
+
+main :: IO ()
+main =
+  do
+    options <- mainGlobal commandLine infoMod
+    case localOptions options of
+        Nothing -> return () -- global options parsed, but local failed; exit gracefully
+        Just KoreFormatOptions {..} ->
+          do
+            defn <- readFile fileName >>= either error return . fromKore fileName
+            let
+              layoutOptions
+                | width > 0 =
+                    defaultLayoutOptions
+                    { layoutPageWidth = AvailablePerLine width 1.0 }
```

Comment:
Indent one more level?

Reply:
As far as I can see, the style guide does not indicate additional indentation here.

Reply:
You're right. When I wrote "Indent everything with multiples of 4 spaces, unless specified otherwise." I wanted to write "Indent everything that is a subcomponent ..." or something like that, but it turns out that I didn't.

After discussing this with @traiansf , this is reasonable so you can leave it as-is. I dislike it a bit since 4 spaces of indentation and a curly brace is a better visual cue than only a curly brace, but that's not the only criteria that matters and I don't have to like everything.

However, there is another interesting issue related to the rule above. In this PR you're using half-indentation for `do` and `let` variables. This may actually be interesting (more interesting for `do`, less interesting for `let` IMO, but others may think differently). However, in the PR you sent to Traian you're using the standard 4 spaces indentation for `do` and `let` variables, so we wondered why they are inconsistent.

So, if you think we should use this half-indentation in some cases (I think you wanted to also use this for `case` statements at the top level of functions), you can submit this as-is, but we should discuss and maybe include these in the style guide doc.

Reply:
> However, there is another interesting issue related to the rule above. In this PR you're using half-indentation for do and let variables. This may actually be interesting (more interesting for do, less interesting for let IMO, but others may think differently).

Sorry, I didn't intend to do that, but old habits die hard. In my personal projects, I have (for years) indented all layout keywords two spaces; when you've done something one way for a long time, your eyes just sail past it without noticing. :smiley:  We can discuss that for the style guide, but for now I will fix this to match what the draft style guide says.

> However, in the PR you sent to Traian you're using the standard 4 spaces indentation for do and let variables, so we wondered why they are inconsistent.

I don't think that PR doesn't actually contains any new code, which probably explains the discrepancy.